File: medium/count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(tree, good, max):
            if tree:
                if tree.val >= max:
                    good.append("")
                    max = tree.val
                dfs(tree.left, good, max)
                dfs(tree.right, good, max)
        res = []
        dfs(root, res, -9e5)
        return len(res)

File: medium/count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.cpp
class Solution {
public:
    void dfs(int &res,int mx,TreeNode* node) {
        if (node) {
            if (node->val>=mx) res++;
            mx=max(mx,node->val);
            dfs(res,mx,node->left);
            dfs(res,mx,node->right);
        }
        return;
    }
    
    int goodNodes(TreeNode* root) {
        int res=0;
        dfs(res,-1e5,root);
        return res;
    }
};
